rules_conditions.md

```diff
@@ -0,0 +1,226 @@
+
+Both the `Value` and the `Datatype` parameters are ignored.
+
+### not-exists
+
+Tests if the specified span does not contain the field named by the `Field` parameter, without considering its value.
+
+Both the `Value` and the `Datatype` parameters are ignored.
+
+### matches
+
+Tests if the span value specified by the `Field` parameter matches the regular expression specified by the `Value` parameter.
+The regular expression grammar used is the syntax used by the Go programming language.
+It is documented [here](https://pkg.go.dev/regexp/syntax).
+
+For clarity, regular expressions in YAML should usually be quoted with single
+quotes (`'`). This is because this form is unambiguous and does not process
+escape sequences, and thus regular expression character classes like `\d` for
+digits can be used directly. For example, an expression to match arbitrary
+strings of digits would be `'\d+'`.
+
+Sometimes double-quoted (`"`) strings are required in order to express patterns
+containing less common characters. These use escape sequences beginning with a
+backslash (`\`). This implies that backslashes intended for the regular
+expression will have to be doubled. The same expression as above using double
+quotes looks like this: `"\\d+"`.
+
+Example:
+```yaml
+      RulesBasedSampler:
+            Rules:
+                - Name: Drop traces for any path element starting with /health or /status
+                  Conditions:
+                    - Field: http.target
+                      Operator: matches
+                      Value: '/(health/status)\w+'
+ ```
+
+Values are always coerced to strings -- the `Datatype` parameter is ignored.
+
+## `Value`
+
+The `Value` parameter can be any value of a supported type.
+Its meaning and interpretation depends on the `Operator` in use.
+
+See [`Datatype`](#datatype) ffor how different datatypes are handled.
+
+## `Datatype`
+
+The `Datatype` parameter controls the type of the values used when evaluating rules for the basic comparison operators.
+When `Datatype` is present, before evaluating the `Operator`, both the span value and the `Value` parameter are coerced (converted if necessary) to the specified format, and the comparison takes place as appropriate for that datatype.
+
+There are 4 possibilities:
+
+- `string` -- the comparison uses string operations (for example, "2" is considered to be greater than "10" because '2' > '1')
+- `int` -- the comparison uses integers (1.5 == 1 because 1.5 gets converted to 1)
+- `float` -- the comparison uses floating point, which generally reflects how most people think of numbers
+- `bool` -- the comparison tries to convert values to boolean. Note that because of the semantics of YAML, the `Value` parameter will interpret not only `true/false` but also all of `yes/no`, `y/n`, and `on/off` as boolean values. Span values, for historical reasons, interpret `true/false` and `1/0` as boolean, and all other values are considered to be `false`.
```

Comment:
```suggestion
- `bool` -- The comparison tries to convert values to boolean. Note that because of the semantics of YAML, the `Value` parameter will interpret not only `true/false` but also all of `yes/no`, `y/n`, and `on/off` as boolean values. Span values, for historical reasons, interpret `true/false` and `1/0` as boolean, and all other values are considered to be `false`.
```